			     BASH PATCH REPORT
			     =================

Bash-Release:	5.3
Patch-ID:	bash53-010

Bug-Reported-by:	Aleksey Covacevice <aleksey.covacevice@gmail.com>
Bug-Reference-ID:
Bug-Reference-URL:	https://lists.gnu.org/archive/html/bug-bash/2026-01/msg00003.html

Bug-Description:

Under some circumstances, a subshell or asynchronous job with an active
EXIT trap that contains a call to `wait' can loop trying to wait for
processes that are not its children. It usually inherits these jobs from
its parent in the jobs list.

--- a/jobs.c
+++ b/jobs.c
@@ -2839,7 +2839,7 @@ wait_for_background_pids (struct procsta
 	  ps->pid = pid;
 	  ps->status = (r < 0 || r > 256) ? 127 : r;
 	}
-      if (r == -1 && errno == ECHILD)
+      if ((r < 0 || r > 256) && errno == ECHILD)
 	{
 	  /* If we're mistaken about job state, compensate. */
 	  check_async = 0;
--- a/patchlevel.h
+++ b/patchlevel.h
@@ -25,6 +25,6 @@
    regexp `^#define[ 	]*PATCHLEVEL', since that's what support/mkversion.sh
    looks for to find the patch level (for the sccs version string). */
 
-#define PATCHLEVEL 9
+#define PATCHLEVEL 10
 
 #endif /* _PATCHLEVEL_H_ */
